templates/frontend/components/registrationFormContexts.tpl

```diff
@@ -26,64 +26,60 @@
-								<div class="custom-control custom-checkbox {if $isSelected}context_privacy_visible{/if}">
-									<input type="checkbox" class="custom-control-input" name="privacyConsent[{$contextId}]" id="privacyConsent[{$contextId}]" value="1"{if $privacyConsent[$contextId]} checked="checked"{/if}>
-									<label for="privacyConsent[{$contextId}]" class="custom-control-label">
-										{capture assign="privacyUrl"}{url router=$smarty.const.ROUTE_PAGE context=$context->getPath() page="about" op="privacy"}{/capture}
-										{translate key="user.register.form.privacyConsentThisContext" privacyUrl=$privacyUrl}
+		<div id="contextOptinGroup" class="registration-context">
+			{foreach from=$contexts item=context}
+				{assign var=contextId value=$context->getId()}
+				{assign var=isSelected value=false}
+				<section>
+					{capture assign="contextUrl"}{url router=$smarty.const.ROUTE_PAGE context=$context->getPath()}{/capture}
+					<h3>
+						<a href="{$contextUrl|escape}" class="registration-context__name">
+							{$context->getLocalizedName()|escape}
+						</a>
+					</h3>
+					<div class="registration-context__roles">
+						<h4>{translate key="user.register.otherContextRoles"}</h4>
+						<div class="custom-control custom-checkbox">
+							{foreach from=$readerUserGroups[$contextId] item=userGroup}
+								{if $userGroup->getPermitSelfRegistration()}
+									{assign var="userGroupId" value=$userGroup->getId()}
+									<input type="checkbox" class="custom-control-input" id="readerGroup[{$userGroupId}]" name="readerGroup[{$userGroupId}]"{if in_array($userGroupId, $userGroupIds)} checked="checked"{/if}>
+									<label for="readerGroup[{$userGroupId}]" class="custom-control-label">
+										{$userGroup->getLocalizedName()}
 									</label>
-								</div>
-							{/if}
-						</li>
-					{/foreach}
-				</ul>
-			</div>
+									{if in_array($userGroupId, $userGroupIds)}
+										{assign var=isSelected value=true}
+									{/if}
+								{/if}
+							{/foreach}
+						</div>
+						<div class="custom-control custom-checkbox">
+							{foreach from=$reviewerUserGroups[$contextId] item=userGroup}
+								{if $userGroup->getPermitSelfRegistration()}
+									{assign var="userGroupId" value=$userGroup->getId()}
+									<input type="checkbox" class="custom-control-input" id="reviewerGroup[{$userGroupId}]" name="reviewerGroup[{$userGroupId}]"{if in_array($userGroupId, $userGroupIds)} checked="checked"{/if}>
+									<label for="reviewerGroup[{$userGroupId}]" class="custom-control-label">
+										{$userGroup->getLocalizedName()}
+									</label>
+									{if in_array($userGroupId, $userGroupIds)}
+										{assign var=isSelected value=true}
+									{/if}
+								{/if}
+							{/foreach}
+						</div>
+					</div>
+					{* Require the user to agree to the terms of the context's privacy policy *}
+					{if !$enableSiteWidePrivacyStatement && $context->getSetting('privacyStatement')}
```

Comment:
Hmm, something prevents this input for consent to be displayed. It may be related to the new changes for the registration form that was introduced in OJS 3.1.2-1. As I remember, they were associated with new variables in a user profile and the consent. Do you want to debug this comparing with templates from the default theme? Let me know if you need help here.

Reply:
Thanks for catching that! I'll debug on my own.

Reply:
I think I’ve fixed it - it doesn’t look great, but I would like to submit a separate PR for properly style and fix the layout of the registration forms: #30!

Reply:
Can you revert the changes in the last commit and add a check here: https://github.com/pkp/pkp-lib/blob/master/templates/frontend/pages/userRegister.tpl#L143 as described in the issue https://github.com/pkp/pkp-lib/issues/4967? This should be enough for now.